pkg/tests/regress_aws_2517_test.go

```diff
@@ -0,0 +1,172 @@
+	testutils "github.com/pulumi/providertest/replay"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+	shim "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim"
+	shimv2 "github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v2"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/tokens"
+)
+
+func TestRegressAws2517(t *testing.T) {
+	ctx := context.Background()
+
+	token := "aws:lb/targetGroup:TargetGroup"
+	tfName := "aws_lb_target_group"
+
+	resource := &schema.Resource{
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		ReadContext: func(_ context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+			m := make(map[string]any)
+			if err := d.Set("target_failover", []any{m}); err != nil {
+				panic(err)
+			}
+			return diag.Diagnostics{}
+		},
+
+		Schema: map[string]*schema.Schema{
+			"target_failover": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"on_deregistration": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"on_unhealthy": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	tfProvider := &schema.Provider{
+		Schema: map[string]*schema.Schema{},
+		ResourcesMap: map[string]*schema.Resource{
+			tfName: resource,
+		},
+	}
+
+	type testCase struct {
+		name         string
+		makeProvider func() shim.Provider
+		interaction  string
+	}
+
+	testCases := []testCase{
+		{
+			name: "default",
+			makeProvider: func() shim.Provider {
+				return shimv2.NewProvider(tfProvider)
+			},
+			interaction: `
+	                {
+			  "method": "/pulumirpc.ResourceProvider/Read",
+			  "request": {
+			    "id": "arn:aws:elasticloadbalancing:us-east-1:616138583583:targetgroup/test-c530e0b/6aee61b47ab16785",
+			    "urn": "urn:pulumi:dev::aws-2517::aws:lb/targetGroup:TargetGroup::foo",
+			    "properties": {}
+			  },
+			  "response": {
+			    "id": "*",
+			    "inputs": "*",
+			    "properties": {
+			       "id": "*",
+			       "targetFailovers": [null]
+			    }
+			  }
+			}`,
+		},
+		{
+			name: "with-flags",
+			makeProvider: func() shim.Provider {
+				return shimv2.NewProvider(tfProvider,
+					shimv2.WithDiffStrategy(shimv2.PlanState),
+					shimv2.WithPlanResourceChange(func(tfResourceType string) bool {
+						return true
+					}),
+				)
+			},
+			interaction: `
+	                {
+			  "method": "/pulumirpc.ResourceProvider/Read",
+			  "request": {
+			    "id": "arn:aws:elasticloadbalancing:us-east-1:616138583583:targetgroup/test-c530e0b/6aee61b47ab16785",
+			    "urn": "urn:pulumi:dev::aws-2517::aws:lb/targetGroup:TargetGroup::foo",
+			    "properties": {}
+			  },
+			  "response": {
+			    "id": "*",
+			    "inputs": "*",
+			    "properties": {
+                               "__meta": "{\"schema_version\":\"0\"}",
```

Comment:
Looks like we just started emitting "0" here instead of skipping zero, but likely no harm done.